clang/test/CIR/Lowering/globals.cir

```diff
@@ -198,4 +205,17 @@ module {
   }
   // MLIR:  %0 = llvm.mlir.addressof @zero_array
 
+  cir.func @const_global_addr() -> !u64i {
+    %0 = cir.const #cir.global_addr<@".str"> : !u64i
```

Comment:
Sounds like this is equivalent to a global view with zero index? Why can't this be modeled ash `cir.const #cir.global_view<@".str"> `?

Reply:
cir.global_addr is of integer types, and I'm not sure if it's appropriate to make cir.global_view an integer type. If it's OK I could just update cir.global_view instead of introducing a new one.

Reply:
My take is that we want here is a constant pointer coming out of `cir.const`, followed up by a cast to integer?

Reply:
> My take is that we want here is a constant pointer coming out of cir.const, followed up by a cast to integer?

In this case, yes. But I'm actually proposing this attribute for the following scenario where you don't have space for additional operations:

```mlir
%0 = cir.const #cir.const_struct {
  #cir.global_addr @aaa,
  #cir.global_addr @bbb
}
```

And I'm looking at this scenario because recently I'm working on the LLVM lowering of member function pointers. Upon ABI lowering, a member function pointer is lowered to a struct with two fields of type `ptrdiff_t`. When the member function pointer represents a non-virtual member function, the first field stores the address of the target function _as an integer_. Thus to represent constant member function pointers I need an attribute that works like `#cir.global_addr`.